.github/workflows/deploy.yml

```
@@ -0,0 +1,19 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  documentation:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Generate documentation
+      run: script/generateLibDocs.sh
+    - name: Deploy documentation
+      uses: peaceiris/actions-gh-pages@v3
```

For security reasons, it's highly recommended to pin this to a commit hash instead of a tag. Otherwise you give a third-party full access to the repository!

---

We need a separate account with limited permissions here. Problem with travis looked much more urgent and we had exact same issue there. Will add it to follow-up tasks. To calm everyone here, those plugins are open source and reviewed for any malicious code before being put to the market place, and if you don't trust third party components in 21-st century, you should implement your own github =)

---

> To calm everyone here, those plugins are open source and reviewed for any malicious code before being put to the market place

That's not true - you can use any github repo as action and anyone with write access to tags can just insert malicious code without any review.

https://julienrenaux.fr/2019/12/20/github-actions-security-risk/

The fix is simple:

```suggestion
      uses: peaceiris/actions-gh-pages@bbdfb200618d235585ad98e965f4aafc39b4c501
```

> and if you don't trust third party components in 21-st century, you should implement your own github =)

Well, this is equivalent to just giving anyone with push access to the action maintainer access to this repo...

---

I do not argue with your point, I've stated that we should ideally have token which cannot do any malicious actions. There are still a lot of `if` in the described case. We also need commit for the `checkout` action. If you could provide PR with those changes, would be great. If not, I will take care of it later today/tomorrow.

---

The `checkout` action is actually reviewed by GitHub, so your argument about trust applies there.

I'll make a PR, but it seems like @foursixnine just put a syntax error into deploy.yml (`You have an error in your yaml syntax on line 13`)

---

We should be all good now, @Vogtinator gogogo!